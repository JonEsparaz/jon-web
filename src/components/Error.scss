.error {
    padding-top: 10vw;
    text-align: center;
    position: relative;
    min-height: 100vh;
    font-family: Heebo;
    background: #1a1a1a;
}

.errortext {
    color: white;
}