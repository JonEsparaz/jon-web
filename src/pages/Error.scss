@import '../theme.scss';

.error {
  padding-top: 5vw;
  text-align: center;
  position: relative;
  height: calc(100vh - 96px - 159px - 5vw);
  font-family: $font;
  background: white;
}
