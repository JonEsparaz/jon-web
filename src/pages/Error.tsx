import React from 'react';
import './Error.scss';
import Menu from '../components/Menu';
import Footer from '../components/Footer';
import { Link } from 'react-router-dom';

export default function Error() {
  return (
    <div>
      <Menu mode="dark" />
      <div className="error">
        <h2 className="Header2 Big">Page not found</h2>
        <Link to="/" className="ActionButton2">
          Back to home
        </Link>
      </div>
      <Footer />
    </div>
  );
}
