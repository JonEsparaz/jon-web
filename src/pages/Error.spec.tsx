import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import Error from './Error';

test('404 error page', () => {
  const history = createBrowserHistory();
  history.push('404');
  const { container, getByText } = render(
    <Router history={history}>
      <Error />
    </Router>,
  );

  expect(container.innerHTML).toMatchSnapshot();

  expect(history.location.pathname).toBe('/404');
  fireEvent.click(getByText('Back to home'));
  expect(history.location.pathname).toBe('/');
});
