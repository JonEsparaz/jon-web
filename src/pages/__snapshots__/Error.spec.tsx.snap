// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 error page 1`] = `"<div class=\\"min-vh-100 d-flex flex-column\\"><div class=\\"px-2 px-sm-5 w-100 position-relative\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo-dark.png\\" class=\\"logo\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider dark-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li class=\\"nav-item\\"><a class=\\"navlink-custom text-decoration-none text-uppercase dark-link\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Speedcubing</span></a></li><li class=\\"nav-item\\"><a class=\\"navlink-custom text-decoration-none text-uppercase dark-link\\" href=\\"/contact\\"><img src=\\"/svg/email-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Contact</span></a></li></ul></div></nav></div><div class=\\"container\\"><div class=\\"row\\"><div class=\\"text-center w-100\\"><h2>Page not found <span role=\\"img\\" aria-label=\\"confused emoji\\">ðŸ˜•</span></h2><a class=\\"action-btn link-btn dark-btn\\" href=\\"/\\"><span class=\\"underline-light\\">Back to home</span></a></div></div></div><div class=\\"flex-grow-1\\"></div><div class=\\"container mt-5 mb-4 text-center\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/twitter.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/github.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/linkedin.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/youtube.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><p class=\\"mt-4\\">Â© 2021 Jonathan Esparaz</p></div></div>"`;
